File: python/day1/day1.py
# read input from file
with open('calories.txt') as f:
    data = f.read().splitlines()

# initialize variables
max_calories = 0
elf_calories = 0

# loop through the list of food Calories
for i in range(len(data)):
    # if the line is empty, reset the current Elf's Calories
    if data[i] == '':
        if elf_calories > max_calories:
            max_calories = elf_calories
        elf_calories = 0
    else:
        # add the current line's Calories to the current Elf's total
        elf_calories += int(data[i])

# check the last Elf's Calories
if elf_calories > max_calories:
    max_calories = elf_calories

# print the result
print("The Elf carrying the most Calories is carrying", max_calories, "Calories.")

File: rust/day1/day1.rs
use std::fs;

fn main() {
    // Read input file into a string
    let input = fs::read_to_string("input.txt").expect("Failed to read input file");

    // Split the string into groups of lines (representing each Elf's inventory)
    let groups: Vec<&str> = input.trim().split("\n\n").collect();

    // Find the Elf with the highest total Calories
    let mut max_calories = 0;
    for group in groups {
        let calories: Vec<u32> = group.lines().map(|s| s.parse().unwrap()).collect();
        let total_calories = calories.iter().sum();
        if total_calories > max_calories {
            max_calories = total_calories;
        }
    }

    // Print the result
    println!("The Elf carrying the most Calories is carrying {} Calories.", max_calories);
}

